compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -396,7 +396,11 @@ class Typer extends Namer
       }
       else if (name.toTermName == nme.ERROR)
         UnspecifiedErrorType
-      else
+      else if (ctx.owner.isConstructor && ctx.owner.owner.unforcedDecls.lookup(tree.name).exists) {
+        // If the field existed but was not found we assume that
+        // we where in the context of an argument of the super constructor
+        errorType(ex"$tree is not accessible from super constructor arguments", tree.pos)
```

Comment:
@smarter do you think this condition is enough?

Reply:
Can't you check `ctx.mode.is(InSuperCall)` ? (The name is misleading, it's set for both this- and super-call, see the method superOrThisCallContext)

Reply:
Also, don't say "super constructor", a this call is not a super call, it's just a constructor call